server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java

```diff
@@ -145,7 +143,19 @@ private SourceFieldMapper(boolean enabled, String[] includes, String[] excludes)
         this.includes = includes;
         this.excludes = excludes;
         final boolean filtered = CollectionUtils.isEmpty(includes) == false || CollectionUtils.isEmpty(excludes) == false;
-        this.filter = enabled && filtered ? XContentMapValues.filter(includes, excludes) : null;
+        if (enabled && filtered) {
+            this.parserConfig = XContentParserConfiguration.EMPTY.withFiltering(Set.of(includes), Set.of(excludes));
```

Comment:
I don't thnk that `parserConfig` needs to be a member variable, it can just be a final local here?

Reply:
Thanks for the suggestions, code updated.